src/Aside/DateCells.tsx

```diff
@@ -35,15 +50,22 @@ const DateCells = ({ date }: DateCellsProps) => {
           totalDays,
           previousMonthTotalDays
         );
-        const cellDate = getCurrentCellDate(firstDayOfMonthInFullDate, day);
-
+        let cellDate = getCurrentCellDate(firstDayOfMonthInFullDate, day);
+        adjustMonthByCellType(cellDate, cellType, day);
```

Comment:
why let?
also this is value mutation, we want to avoid mutation as much as we can
if you need adjustments to be made create new variable

```suggestion
        const currentCellDate = getCurrentCellDate(firstDayOfMonthInFullDate, day);
        const adjustedDateByCellType = getAdjustedDateByCellType(cellDate, cellType, day);
```